### ISSUE TITLE
E2E test intermittent failures due to account not found

### ISSUE THREAD

**Issue by chatton** [cosmos/ibc-go#4751]
## Summary of Bug

I noticed the `TestScheduleIBCUpgrade_Succeeds` has become flakey.  [example failure](https://github.com/cosmos/ibc-go/actions/runs/6272041965/job/17032990347?pr=4737)

I have seen the following error several times

```
=== RUN   TestClientTestSuite/TestScheduleIBCUpgrade_Succeeds/execute_proposal_for_MsgIBCSoftwareUpgrade
=== NAME  TestClientTestSuite/TestScheduleIBCUpgrade_Succeeds
    tx.go:63: 
        	Error Trace:	/home/runner/work/ibc-go/ibc-go/e2e/testsuite/tx.go:63
        	            				/home/runner/work/ibc-go/ibc-go/e2e/testsuite/tx.go:161
        	            				/home/runner/work/ibc-go/ibc-go/e2e/tests/core/02-client/client_test.go:115
        	Error:      	Received unexpected error:
        	            	rpc error: code = NotFound desc = rpc error: code = NotFound desc = account cosmos1qsklq72jkfypur82f407nhhj886548jttcg399 not found: key not found
```

A re-run tends to solve the issue.

## Expected Behaviour

Test passes consistently.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged/assigned

**Comment by crodriguezvega** [comment]
Same thing for [`TestClient_Update_Misbehaviour`](https://github.com/cosmos/ibc-go/actions/runs/6273050262/job/17035783233?pr=4658):

```
Error Trace:	/home/runner/work/ibc-go/ibc-go/e2e/testsuite/tx.go:63
[216](https://github.com/cosmos/ibc-go/actions/runs/6273050262/job/17035783233?pr=4658#step:5:217)
        	            				/home/runner/work/ibc-go/ibc-go/e2e/tests/core/02-client/client_test.go:312
[217](https://github.com/cosmos/ibc-go/actions/runs/6273050262/job/17035783233?pr=4658#step:5:218)
        	Error:      	Received unexpected error:
[218](https://github.com/cosmos/ibc-go/actions/runs/6273050262/job/17035783233?pr=4658#step:5:219)
        	            	rpc error: code = NotFound desc = rpc error: code = NotFound desc = account cosmos1ag9sfaa0m70m7pzdlw38cvmy37kpdu5axrn8x4 not found: key not found
[219](https://github.com/cosmos/ibc-go/actions/runs/6273050262/job/17035783233?pr=4658#step:5:220)
        	Test:       	TestClientTestSuite/TestClient_Update_Misbehaviour
```

**Comment by DimitrisJim** [comment]
A number of these are flaky. It's an error I remember seeing when adding hermes. Making hermes snooze for a bit longer is what usually did the trick.

**Comment by chatton** [comment]
Perhaps we could add a `WaitUntil(addressFound)` type functionality here instead of sleeping longer? Or add to the list of [retry messages](https://github.com/cosmos/ibc-go/blob/585f9944c780349395d4845014a5874599c0b542/e2e/testsuite/tx.go#L74).

**Comment by DimitrisJim** [comment]
Actually bumped into Luca from Informal that works on hermes and mentioned this to him. Going to coordinate a bit with him on slack to see if this is something that can be fixed on hermes.

**Comment by chatton** [comment]
I've also actually seen this issue in the V8 upgrade test also with the same error (I think also just with hermes, but not 100% sure on that, something to keep an eye out for)

**Comment by chatton** [comment]
I think this has been resolved by #6689 if we see this error again, I think we can either open this issue again or create a new one with more recent examples.
